XenModel/Actions/Host/FibreChannelProbeAction.cs

```diff
@@ -71,14 +67,47 @@ protected override void Run()
                     else
                         throw;
                 }
+
+                FibreChannelDevices = ProcessXml(Result);
             }
             else
             {
                 var deviceConfig = new Dictionary<string, string>();
                 deviceConfig["provider"] = "hba";
-                ProbeExtResult = SR.probe_ext(Session, Host.opaque_ref, deviceConfig, srType.ToString(), new Dictionary<string, string>());
+                var result = SR.probe_ext(Session, Host.opaque_ref, deviceConfig, srType.ToString(), new Dictionary<string, string>());
+                FibreChannelDevices = (from Probe_result r in result select new FibreChannelDevice(r.extra_info)).ToList();

```

Comment:
The "ScsiId" value is returned in probeResult.configuration, not in probeResult.extra_info